src/libp2p_multistream.erl

```diff
@@ -23,23 +23,28 @@ write_lines(Connection, Lines) ->
 
 -spec read(libp2p_connection:connection()) -> string() | {error, term()}.
 read(Connection) ->
-    case read_varint(Connection) of
+    read(Connection, 60000).
+
+-spec read(libp2p_connection:connection(), pos_integer()) -> string() | {error, term()}.
+read(Connection, Timeout) ->
+    case read_varint(Connection, Timeout) of
         {error, Error} ->
             {error, Error};
         {ok, Size} when Size > ?MAX_LINE_LENGTH ->
             {error, {line_too_long, Size}};
         {ok, Size} ->
-            case libp2p_connection:recv(Connection, Size) of
+            case libp2p_connection:recv(Connection, Size, Timeout) of
                 {error, Error} -> {error, Error};
                 {ok, <<Data:Size/binary>>} ->
                     {Line, <<>>} = decode_line_body(Data, Size),
                     Line
             end
     end.
 
+
 -spec read_lines(libp2p_connection:connection()) -> [string()] | {error, term()}.
 read_lines(Connection) ->
-    case read_varint(Connection) of
+    case read_varint(Connection, 60000) of
```

Comment:
Declare a RECV_TIMEOUT and use that macro here to be consistent with other locations